### PR TITLE
[FX-799] Expose deferPaymentCapture

### PR DESCRIPTION
**Pull request by dleis612**
## What

- https://linear.app/joinforage/issue/FX-799/ios-implement-the-collectpin-function

## Test Plan
~I plan on adding unit tests if possible and will add to the sample app.~

- [x] Added unit tests
- [x] #141 introduces the necessary sample app changes needed to implement QA tests

## 

Can be released as-is, but should be followed by this sample app diff as well #141 AND #143

### CODE REVIEW

File: Sources/ForageSDK/Foundation/Network/LiveForageService.swift [hunk 1]
```diff
@@ -133,6 +134,67 @@ class LiveForageService: ForageService {
...
+    }
+
+    /// Common Payment-related prologue across capturePayment and collectPin.
+    /// Both `collectPin` and `capturePayment` involve the same
+    /// preliminerary data retrieval and a trip to the Vault (VGS or Basis Theory) Proxy
+    private func collectPinForPayment(
+        pinCollector: VaultCollector,
+        paymentReference: String,
+        action: VaultAction
+    ) async throws -> (vaultResponse: VaultResponse, forageRequest: ForageRequestModel) {
```
Comment (djoksimo) [inline Sources/ForageSDK/Foundation/Network/LiveForageService.swift:194]: I want to eventually deprecate our use of `ForageRequestModel`

See ticket here 👉🏻 
https://linear.app/joinforage/issue/FX-650/refactor-foragerequestmodel-empty-strings

File: Sources/ForageSDK/Foundation/Network/LiveForageService.swift [hunk 2]
```diff
@@ -78,6 +78,7 @@ class LiveForageService: ForageService {
 
             let vaultResult = try await submitPinToVault(
                 pinCollector: pinCollector,
+                vaultAction: VaultAction.balanceCheck,
```
Comment (dleis612) [inline Sources/ForageSDK/Foundation/Network/LiveForageService.swift:81]: Can change to `.balanceCheck`

**Review by dleis612 — COMMENTED** [review]
@djoksimo LGTM! I can't +1 because it's my PR

File: Sources/ForageSDK/Foundation/Network/LiveForageService.swift [hunk 3]
```diff
@@ -133,6 +134,67 @@ class LiveForageService: ForageService {
...
+
+    func collectPinForDeferredCapture(
+        pinCollector: VaultCollector,
+        paymentReference: String
+    ) async throws -> VaultResponse {
+        do {
+            let collectPinResult = try await collectPinForPayment(
+                pinCollector: pinCollector,
+                paymentReference: paymentReference,
+                action: .collectPin
```
Comment (dleis612) [inline Sources/ForageSDK/Foundation/Network/LiveForageService.swift:179]: Adjust to `.deferCapture`.

File: Sources/ForageSDK/Foundation/Network/Collector.swift [hunk 4]
```diff
@@ -13,6 +13,24 @@ import VGSCollectSDK
 let tokenDelimiter = ","
 let tokenKey = "card_number_token"
 
+/// Defines actions performed against the Vault (VGS or Basis Theory) Proxy.
+enum VaultAction: String {
+    case balanceCheck = "balance"
+    case capturePayment = "capture"
+    case collectPin = "collect_pin"
```
Comment (dleis612) [inline Sources/ForageSDK/Foundation/Network/Collector.swift:20]: `case deferCapture = "defer_capture"`

File: Sources/ForageSDK/Foundation/Network/LiveForageService.swift (hunk 3 above)
Comment (djoksimo) [reply Sources/ForageSDK/Foundation/Network/LiveForageService.swift]: c65c391da582315b639a0159fe41487bf5b24e3d

File: Sources/ForageSDK/Foundation/Network/Collector.swift (hunk 4 above)
Comment (djoksimo) [reply Sources/ForageSDK/Foundation/Network/Collector.swift]: c65c391da582315b639a0159fe41487bf5b24e3d

**Review by djoksimo — APPROVED** [review]

**Pull request merged**